Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java

```diff
@@ -339,6 +342,16 @@ protected void execute() throws InterruptedException, ExecutionException {
         captureYield(harvestedBlocks);
     }
 
+    protected void updateTrackedLocations(Craft craft, int dx, int dy, int dz) {
+        craft.getTrackedLocations().values().forEach(trackedLocations -> {
+            trackedLocations.forEach(
+                    trackedLocation -> {
+                        trackedLocation.translate(dx, dy, dz);
+                    }
+            );
+        });
+    }
+
```

Comment:
Tracked locations do not need to be translated.  They are stored as an offset from the center of a craft.

Reply:
The overhead is addition/subtraction operations, neither is very expensive and the overhead was not noticeable on a 88k craft was ~2k trackedlocations during tests.

in addition to that, that is required. As of now the craft‘s center changes during the craft‘s existence. When a subcraft is created, a torpedo launched or skiff launched or on rotation (when at least one side is of even length) the centerpoint will move somewhere else resulting in the vector, which does not change, pointing to the wrong location. To fix that, we need a event or callback to react to the hitbox changing. Will try something later